DOC: Proofread geneve.rst (#1754)
vyos/vyos-documentation: docs/configuration/interfaces/geneve.rst

:lastproofread: 2026-02-02

.. _geneve-interface:

######
Geneve
######

:abbr:`Geneve (Generic Network Virtualization Encapsulation)` interfaces 
operate as virtual network ports. Administrators can apply standard network 
configurations on them, such as IP addressing, bridging, or firewall rules, 
just as they would on physical Ethernet ports.

> To transport data across the network, they utilize the Geneve encapsulation protocol. This 

Reviewer: ```suggestion
This 
```

A completely redundant phrase.

Author: Removed.

GENEVE
######

> :abbr:`GENEVE (Generic Network Virtualization Encapsulation)` interfaces 

Reviewer: A purely aesthetic point: the [RFC that defines the protocol](https://www.rfc-editor.org/rfc/rfc8926) and other IETF documents seem to consistently use "Geneve" styling rather than "GENEVE". We may want to follow that.

Author: Corrected.

operate as virtual network ports. Administrators can apply standard network 
configurations on them, such as IP addressing, bridging, or firewall rules, 
just as they would on physical Ethernet ports.

GENEVE interfaces are also used for configuring GENEVE tunnels. To transport 

> data across the network, they utilize the GENEVE encapsulation protocol. This 

Reviewer: ```suggestion
To transport  data across the network, they utilize the GENEVE encapsulation protocol. This 
```

Not "also" — they are called `geneve` because their sole purposes is to create Geneve tunnels, as the name implies. ;)

protocol encapsulates Layer 2 Ethernet frames originating from endpoints such 
as virtual machines, containers, or physical servers inside UDP packets. It 
unifies the features of earlier encapsulation protocols, including VXLAN, 
NVGRE, and STT, and addresses their limitations, such as fixed header 
structures and a lack of metadata support. Because of its extensibility, GENEVE 

> is considered the successor to these legacy protocols.

Reviewer: ```suggestion
structures and a lack of metadata support. Because of its extensibility, GENEVE 
may eventually replace those older protocols.
```

I wouldn't call VXLAN a legacy protocol. STT and NVGRE have indeed mostly lost vendor support now (they were always single-vendor protocols of MS and VMware) but VXLAN is widely supported and isn't going anywhere anytime soon.

  :var0: geneve
  :var1: gnv0

Geneve options
==============

.. cfgcmd:: set interfaces geneve gnv0 remote <address>

   Configure the remote endpoint IP address for the Geneve tunnel.

.. cfgcmd:: set interfaces geneve gnv0 vni <vni>

   **Configure** :abbr:`VNI (Virtual Network Identifier)` **for the Geneve 
   interface.**

   The VNI is a virtual network identifier. It allows multiple virtual networks to 
   share the same physical infrastructure and remain isolated.

   The VNI is also used to distribute traffic after it leaves the tunnel, for 
   example, to map packets with overlapping IP addresses to specific routing 

>    tables or to distribute traffic across multiple CPU cores.

Reviewer: ```suggestion
   tables.
```

Where does that come from? I don't think Linux ever uses anything inside packets to decide what CPU to handle it with — for physical interfaces it's sometimes possible to map interrupt numbers to CPUs but for virtual interfaces there's no way to influence it at all (neither one is needed).

Author: Removed.

.. cfgcmd:: set interfaces geneve gnv0 remote <address>

   Configure the remote endpoint IP address for the GENEVE tunnel.

.. cfgcmd:: set interfaces geneve gnv0 vni <vni>

   **Configure** :abbr:`VNI (Virtual Network Identifier)` **for the GENEVE 
   interface.**

   The VNI is a virtual network identifier. It allows multiple virtual networks to 

>    share the same physical infrastructure while remaining isolated.

Reviewer: ```suggestion
   share the same physical infrastructure and remain isolated.
```

I think in most cases verbs read better than gerunds.
